docker/src/main/java/org/jclouds/docker/config/DockerHttpApiModule.java

```diff
@@ -48,8 +65,45 @@ protected void bindErrorHandlers() {
    @Override
    protected void configure() {
       super.configure();
-      install(new OkHttpCommandExecutorServiceModule());
-      bind(OkHttpClientSupplier.class).to(DockerOkHttpClientSupplier.class);
+      install(new SSLModule());
+      bind(HttpCommandExecutorService.class).to(OkHttpCommandExecutorService.class).in(Scopes.SINGLETON);
+      bind(OkHttpClient.class).toProvider(DockerOkHttpClientProvider.class).in(Scopes.SINGLETON);
+  }
+
+   private static final class DockerOkHttpClientProvider implements Provider<OkHttpClient> {
+      private final HttpUtils utils;
+      private final Supplier<SSLContext> sslContextWithKeysSupplier;
+      private final OkHttpClientSupplier clientSupplier;
+      private final HostnameVerifier hostnameVerifier;
+
+      @Inject
+      DockerOkHttpClientProvider(OkHttpClientSupplier clientSupplier, HttpUtils utils, SSLContextWithKeysSupplier sslContextWithKeysSupplier, @Named("untrusted") HostnameVerifier hostnameVerifier) {
+         this.clientSupplier = clientSupplier;
+         this.utils = utils;
+         this.sslContextWithKeysSupplier = sslContextWithKeysSupplier;
+         this.hostnameVerifier = hostnameVerifier;
+      }
+
+      @Override
+      public OkHttpClient get() {
+         OkHttpClient client = clientSupplier.get();
+         client.setConnectTimeout(utils.getConnectionTimeout(), TimeUnit.MILLISECONDS);
+         client.setReadTimeout(utils.getSocketOpenTimeout(), TimeUnit.MILLISECONDS);
+         client.setFollowRedirects(false);
+         ConnectionSpec tlsSpec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+                 .tlsVersions(TlsVersion.TLS_1_0, TlsVersion.TLS_1_1, TlsVersion.TLS_1_2)
+                 .build();
+         ConnectionSpec cleartextSpec = new ConnectionSpec.Builder(ConnectionSpec.CLEARTEXT)
+                 .build();
+         client.setConnectionSpecs(ImmutableList.of(tlsSpec, cleartextSpec));
+         client.setSslSocketFactory(sslContextWithKeysSupplier.get().getSocketFactory());
+
+         if (utils.relaxHostname()) {
+            client.setHostnameVerifier(hostnameVerifier);
+         }
+         return client;
+      }
```

Comment:
Agree. The trustallcerts thing is something like "Oh, ok, I don't want to deal (and I don't care) with SSL at all but I have an SSL endpoint", so I'd say the Docker certs should also be ignored. If that is the case, the default implementation of the untrusted supplier should already work and no changes would be required.